zulipterminal/ui_tools/boxes.py

```diff
@@ -104,17 +104,22 @@ def generic_autocomplete(self, text: str, state: int) -> Optional[str]:
             if text.startswith(prefix):
                 typeaheads = autocomplete_func(text, prefix)
                 reduced_typeaheads = typeaheads[:num_suggestions]
+                is_truncated = len(reduced_typeaheads) != len(typeaheads)
 
                 if state in range(-len(reduced_typeaheads),
                                   len(reduced_typeaheads)):
-                    return reduced_typeaheads[state]
+                    typeahead = reduced_typeaheads[state][0]
                 else:
-                    return None
+                    typeahead = None  # type: ignore
```

Comment:
We return an `Optional`, so what's the problem here?

Reply:
Fixed.

Reply:
The type is not determined by the return type, but by 
```
typeahead = reduced_typeaheads[state]
```

Reply:
Yeah, adding a type for `typeahead = reduced_typeaheads[state]` would be ideal but it does not fit the line